From 7202c3af81c208521524389fcf374fd73211b51b Mon Sep 17 00:00:00 2001
From: Ran Benita <ran@unusedvar.com>
Date: Wed, 27 May 2026 11:03:05 +0300
Subject: [PATCH] testing: fix `test_dont_write_bytecode` for Python 3.15

Python 3.15b1 no longer sets `module.__cached__`. But
`module.__spec__.cached` is still set, is recommended over
`module.__cached__` by Python documentation, and it works for all
previous versions as well, so switch to that.

Ref:
https://docs.python.org/3.15/reference/datamodel.html#module.__file__
---
 testing/test_assertrewrite.py | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/testing/test_assertrewrite.py b/testing/test_assertrewrite.py
index 2668001af65..e11863547ba 100644
--- a/testing/test_assertrewrite.py
+++ b/testing/test_assertrewrite.py
@@ -1082,9 +1082,10 @@ def test_dont_write_bytecode(self, pytester: Pytester, monkeypatch) -> None:
             """
             import os
             def test_no_bytecode():
-                assert "__pycache__" in __cached__
-                assert not os.path.exists(__cached__)
-                assert not os.path.exists(os.path.dirname(__cached__))"""
+                assert "__pycache__" in __spec__.cached
+                assert not os.path.exists(__spec__.cached)
+                assert not os.path.exists(os.path.dirname(__spec__.cached))
+            """
         )
         monkeypatch.setenv("PYTHONDONTWRITEBYTECODE", "1")
         assert pytester.runpytest_subprocess().ret == 0
